compiler/wasm/test/node/index.test.ts

```diff
@@ -1,26 +1,72 @@
 import { expect } from 'chai';
-import { nargoArtifactPath, noirSourcePath } from '../shared';
+import {
+  depsScriptSourcePath,
+  depsScriptExpectedArtifact,
+  libASourcePath,
+  libBSourcePath,
+  simpleScriptSourcePath,
+  simpleScriptExpectedArtifact,
+} from '../shared';
 import { readFileSync } from 'node:fs';
-import { join } from 'node:path';
+import { join, resolve } from 'node:path';
 import { compile } from '@noir-lang/noir_wasm';
+import { initializeResolver } from '@noir-lang/source-resolver';
 
-const absoluteNoirSourcePath = join(__dirname, noirSourcePath);
-const absoluteNargoArtifactPath = join(__dirname, nargoArtifactPath);
+// const absoluteNoirSourcePath = ;
+// const absoluteNargoArtifactPath = join(__dirname, nargoArtifactPath);
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-async function getPrecompiledSource(): Promise<any> {
-  const compiledData = readFileSync(absoluteNargoArtifactPath).toString();
+async function getPrecompiledSource(path: string): Promise<any> {
+  const compiledData = readFileSync(resolve(__dirname, path)).toString();
   return JSON.parse(compiledData);
 }
 
 describe('noir wasm compilation', () => {
-  it('matches nargos compilation', async () => {
-    const wasmCircuit = await compile(absoluteNoirSourcePath);
-    const cliCircuit = await getPrecompiledSource();
-
-    // We don't expect the hashes to match due to how `noir_wasm` handles dependencies
-    expect(wasmCircuit.bytecode).to.eq(cliCircuit.bytecode);
-    expect(wasmCircuit.abi).to.deep.eq(cliCircuit.abi);
-    expect(wasmCircuit.backend).to.eq(cliCircuit.backend);
-  }).timeout(10e3);
+  describe('can compile simple scripts', () => {
+    it('matching nargos compilation', async () => {
+      const wasmCircuit = await compile(join(__dirname, simpleScriptSourcePath));
+      const cliCircuit = await getPrecompiledSource(simpleScriptExpectedArtifact);
+
+      // We don't expect the hashes to match due to how `noir_wasm` handles dependencies
+      expect(wasmCircuit.bytecode).to.eq(cliCircuit.bytecode);
+      expect(wasmCircuit.abi).to.deep.eq(cliCircuit.abi);
+      expect(wasmCircuit.backend).to.eq(cliCircuit.backend);
+    }).timeout(10e3);
+  });
+
+  describe('can compile scripts with dependencies', () => {
+    beforeEach(() => {
+      initializeResolver((file) => {
+        switch (file) {
+          case '/script/main.nr':
+            return readFileSync(join(__dirname, depsScriptSourcePath), 'utf-8');
+
+          case '/lib_a/lib.nr':
+            return readFileSync(join(__dirname, libASourcePath), 'utf-8');
+
+          case '/lib_b/lib.nr':
+            return readFileSync(join(__dirname, libBSourcePath), 'utf-8');
+
+          default:
+            return '';
+        }
+      });
```

Comment:
Making a note here that we're forced into using a custom resolver here due to the fact we're not differentiating between package names and their paths. Not a blocker for this PR however.

Reply:
I've added a comment in the `it` portion of the test.